### ISSUE TITLE
World Drop update

### ISSUE THREAD

**Issue by Felina-Lain** [Domochevsky/minecraft-worlddrop#1]
Hello 

It was a bit of a guessing game but I think I got everything, like you said it was mainly new names for functions that you had written already. I did some guessing using the arguments, I figured the functions would still have a similar name to the old one and with the same arguments.

I went and did that for the other scripts as well, using the arguments again to guess the new functions names or new types names.

After all this I'm left with only one error, in the Transition script:

 Entity newEntity = EntityList.createEntityByName(EntityList.getEntityString(entity), wsNew);
 "The method createEntityByName(String, WorldServer) is undefined for the type EntityList"
 

I tried guessing again, but I couldn't find a function that take for arguments a string and a world server. The closest I found is

createEntityByIDFromName(name, worldIn)

But the name argument is a Resource location and worldIn is just a world and not a world server
 

Hope you're having a nice day, and thanks once again

**Comment by Domochevsky** [comment]
Gud gud.

Hum... is there anything in entity that gives you back its ResourceLocation? Like entity.getResourceLocation()? Or in Entity/EntityList as a static class with the entity as argument? Like EntityList.getResourceLocationFromEntity(entity). 

(The fuck kind of function name is **createEntityByIDFromName** anyway? Is it "by ID" or "from Name"?)

WorldServer is a subtype of World meanwhile, if I remember correctly, so that should work just fine as argument.

**Comment by Felina-Lain** [comment]
EntityList has this EntityList.getKey(entityIn) which is described as 
"Gets the ResourceLocation that identifies the given entity's type.Parameters:entityIn
Returns:The resource location, or null if the REGISTRY does not contain a mapping for that class."

Went ahead, and gave it a try, it's working!
Tested some of the commands, seems working, I just jumped into the nether from height 60 XD

Thank you so so much for your help! This is awesome! \^.^/
(I'll probably upload the update in the coming week, if you want the new files!)

**Comment by Domochevsky** [comment]
Noice. I do want the new files, yes. Maybe do a branch specifically for 1.12 and then pull request that over here. :)

(ow with CurseForge out of the picture, the mod jar files should probably be hosted here as well.)

**Comment by Felina-Lain** [comment]
EDIT: got it, I never forked before, that was why XD

And I got a CurseForge/twitch account so if you don't mind it, I post the 1.12 jar there, with a link to the old versions as well :)

**Comment by Domochevsky** [comment]
Yeah, that should be fine. Don't forget to also comment on it with a link to the new place so people know where to go.

A thing that should be done is uploading the jar here too, for general easy of access. (So it sits in more places than just CurseForge. That can then be linked to in the readme, since that's what shows first when this repository is opened.)

**Comment by Felina-Lain** [comment]
Alright, I'll make the curseforge page, then send you a link to dl the jar directly if you need

I'll copy the description from your thread, and the icon as well, unless you want me to find new ones

**Comment by Domochevsky** [comment]
Coo', that should work. :)

Oh yeah, don't forget to link to the github repository as well. The mod is open source after all, so any fork from it has to be as well and marked as such.
